virttest/qemu_storage.py

```diff
@@ -986,9 +986,12 @@ def convert(self, params, root_dir, cache_mode=None,
         convert_cmd = self.image_cmd + " " + \
             self._cmd_formatter.format(self.convert_cmd, **cmd_dict)
 
-        LOG.info("Convert image %s from %s to %s", self.image_filename,
-                 self.image_format, convert_image.image_format)
-        process.run(convert_cmd)
+        timeout = params.get_numeric("conversion_timeout", 12000)
```

Comment:
Also, I'd suggest setting *no timeout* as the default value if we plan to support it, which is the current behavior.

Reply:
How about a very large number as a default value? Can you think of practical use cases where waiting infinitely long would make sense?

Reply:
As [*test timeout*](https://avocado-framework.readthedocs.io/en/latest/guides/writer/chapters/writing.html#setting-a-test-timeout) will take effect eventually I believe it won't be really waiting "infinitely" long.

From my POV, the critical issues were:

* how to set a rational default timeout for each single operation of a test?
* how to handle the relationship between that and test timeout?

since how long a image conversion would take depends on a couple of factors (image size, hardware storage performance, system load status, etc.), it's hard for me to say if 12000s as the default is better or not comparing with using the test timeout for the **most** cases. What if someone would request to enlarge/reduce the number in future, how to evaluate that? I can't answer that this moment, so I just suggested what I put in the previous comment. I think users know their case better than us so I'd like to let them handle the thing independently.

Reply:
Indeed, see above for the solution I would push.